phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -4168,4 +4176,197 @@ public MutationState useSchema(UseSchemaStatement useSchemaStatement) throws SQL
+            grantPermissionsToTable(hConnection, grantStatement, tableName);
+        }
+
+        byte[] viewIndexTableBytes = MetaDataUtil.getViewIndexPhysicalName(inputTable.getPhysicalName().getBytes());
+        tableName = org.apache.hadoop.hbase.TableName.valueOf(viewIndexTableBytes);
+        boolean viewIndexTableExists = connection.getQueryServices().getAdmin().tableExists(tableName);
+        if(!viewIndexTableExists && inputTable.isMultiTenant()) {
+            logger.error("View Index Table not found for MultiTenant Table: " + inputTable.getName());
+            throw new TablesNotInSyncException(inputTable.getTableName().getString(),
+                    Bytes.toString(viewIndexTableBytes), " View Index table should exist for MultiTenant tables");
+        }
+        if(viewIndexTableExists) {
+            grantPermissionsToTable(hConnection, grantStatement, tableName);
+        }
+
+    }
+
+    private void grantPermissionsToTable(HConnection hConnection, GrantStatement grantStatement, org.apache.hadoop.hbase.TableName tableName)
+            throws Throwable {
+        AccessControlClient.grant(hConnection, tableName, grantStatement.getName(),
+                null, null, grantStatement.getPermsList());
+    }
+
+    private void grantPermissionsToSchema(HConnection hConnection, GrantStatement grantStatement)
+            throws Throwable {
+        AccessControlClient.grant(hConnection, grantStatement.getSchemaName(), grantStatement.getName(), grantStatement.getPermsList());
+    }
+
+    private void grantPermissionsToUser(HConnection hConnection, GrantStatement grantStatement)
+            throws Throwable {
+        AccessControlClient.grant(hConnection, grantStatement.getName(), grantStatement.getPermsList());
+    }
+
+    public MutationState revokePermission(RevokeStatement revokeStatement) throws SQLException {
+
+        StringBuffer revokePermLog = new StringBuffer();
+        revokePermLog.append("Revoke Permissions requested for user/group: " + revokeStatement.getName());
+        if (revokeStatement.getSchemaName() != null) {
+            revokePermLog.append(" for Schema: " + revokeStatement.getSchemaName());
+        } else if (revokeStatement.getTableName() != null) {
+            revokePermLog.append(" for Table: " + revokeStatement.getTableName());
+        }
+        logger.info(revokePermLog.toString());
+
+        HConnection hConnection = connection.getQueryServices().getAdmin().getConnection();
+
+        try {
+            if (revokeStatement.getSchemaName() != null) {
+                // SYSTEM.CATALOG doesn't have any entry for "default" HBase namespace, hence we will bypass the check
+                if(!revokeStatement.getSchemaName().equals(QueryConstants.HBASE_DEFAULT_SCHEMA_NAME)) {
+                    FromCompiler.getResolverForSchema(revokeStatement.getSchemaName(), connection);
```

Comment:
Does FromCompiler.getResolverForSchema actually check to see if the schema exists?

Reply:
Yes, it tries to resolve the schema immediately and throws an exception if Schema is not found. 
`return new SchemaResolver(connection, SchemaUtil.normalizeIdentifier(schema), updateCacheImmediately: true);`